Source/DataStructureAndEncodingDefinition/fixed_istream_buffer.h

```diff
@@ -0,0 +1,86 @@
+#include <streambuf>
+#include <cstring>
+#include <vector>
+#include <istream>
+
+/**
+ The gdcmReader does seeking when decoding the headers so we need to buffer the input stream.
+ You can seek back at least half the buffer size and at most the full buffer size depending on the sequence position.
+ You may not be able to seek forward but can seek forward at most half the buffer size depending on the sequence position.
```

Comment:
GDCM may seek forward when skipping tags and so it is possible to fail in this case.
There is also a case for implicit transfer syntax's where it will try to seek to the end to determine the read length and this will fail as well.